core/src/test/java/com/amplifyframework/hub/HubEventDataObjectsTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.hub;
+
+import com.amplifyframework.api.events.ApiEndpointStatusChangeEvent;
+import com.amplifyframework.api.events.ApiEndpointStatusChangeEvent.ApiEndpointStatus;
+import com.amplifyframework.datastore.events.NetworkStatusEvent;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+
+/**
+ * Test basic operations of objects used as Hub event payloads.
+ */
+@RunWith(RobolectricTestRunner.class)
+public class HubEventDataObjectsTest {
+
+    /**
+     * Verify {@link NetworkStatusEvent} behavior.
+     */
+    @Test
+    public void verifyNetworkStatus() {
+        NetworkStatusEvent status1 = new NetworkStatusEvent(true);
+        NetworkStatusEvent status2 = new NetworkStatusEvent(false);
+        NetworkStatusEvent status3 = new NetworkStatusEvent(true);
+
+        assertNotEquals(status1, status2);
+        assertEquals(status1, status3);
+
+        assertNotEquals(status1.hashCode(), status2.hashCode());
+        assertEquals(status1.hashCode(), status3.hashCode());
+        assertFalse(status1 == status3);
+
+        assertNotEquals(status1.toString(), status2.toString());
+        assertEquals(status1.toString(), status3.toString());
```

Comment:
Personally I agree with you that these are useful types of tests. However, I recently changed _how_ I do this test, of the `hashCode()`. Rarely will anyone ever call `hashCode()`. Perhaps they will if they are writing _their own_, higher level `hashCode()`, and include one of these types in it. But more often, they'll probably try to use it in a collection. So like:
```java
PossibleSaneHashObject one = new PossiblySaneHashObject();
PossibleSaneHashObject two = new PossiblySaneHashObject();
Set<PossiblySaneHashObject> set = new HashSet<>();
set.add(one);
set.add(two);
assertEquals(1, set.size());
```

This is straight-up pedantry at this point, feel free to ignore 😆

Reply:
It makes sense...especially as we start adding more events down the road. That being said, the test itself seems pretty generic, so I created a class under tesutils called `ObjectValidatorUtils`. Right now it has one method that takes 3 objects: object1, object2, object3. The objects must be constructed such that:

object1.equals(object3), but not ==.
object2 is different from object1 and object3.

Let me know if this makes sense. Naming stuff is definitely not my superpower 🤣